src/app/core/services/data.service.ts

```diff
@@ -62,9 +63,16 @@ export class DataService {
     this.subject.next(this.love ? lovedVideos : this.userVideosList);
   }
 
-  public deleteVideo(id: string): void {
-    this.userVideosList = this.userVideosList.filter((video) => video.videoId !== id);
-    this.subject.next(this.userVideosList);
+  public deleteVideo(videoId: string): void {
+    this.lastVideo = this.userVideosList.filter((video, index) => {
+      if(video.videoId === videoId){
+          video.id = index
+          return true
+      }
+      return false
+    });
```

Comment:
hmm a może coś takiego 
this.lastDeletedVideo = this.userVideosList.find(video => video.videoId === videoId)
i potem przy undo po prostu mógłbyś wrzucić to video 
this.userVideosList.push(this.lastDeletedVideo)

Reply:
Usprawniłem kod.